packages/devtools_flutter/lib/src/screen.dart

```diff
@@ -4,119 +4,58 @@
 
 import 'package:flutter/material.dart';
 
-/// A screen in the DevTools App, including the scaffolding and navigation tabs
-/// for navigating the app.
-///
-/// This widget is used by encapsulation instead of inheritance, so to add a
-/// FooScreen to the app, you'll create a FooScreen widget like so:
-///
-/// ```dart
-/// class FooScreen extends StatelessWidget {
-///   @override
-///   Widget build(BuildContext context) {
-///     return Screen(
-///       child: /* Build out the screen content */,
-///     )
-///   }
-/// }
-/// ```
-///
-/// For a sample implementation, see [ConnectScreen].
-class Screen extends StatefulWidget {
-  const Screen({Key key, @required this.child})
-      : assert(child != null),
-        super(key: key);
-
-  static const Key narrowWidth = Key('Narrow Screen');
-  static const Key fullWidth = Key('Full-width Screen');
-
-  /// The width where we need to treat the screen as narrow-width.
-  static const double narrowScreenWidth = 800.0;
-
-  final Widget child;
-
-  @override
-  State<StatefulWidget> createState() => ScreenState();
+/// Defines pages shown in the tabbar of the app.
+@immutable
+abstract class Screen {
+  const Screen(this.name);
+
+  /// The human-readable name to show for the app.
```

Comment:
nit: r/app/screen

Reply:
Done.  I've also corrected some of the documentation in here.